src/torsion_profile.py

```diff
@@ -0,0 +1,191 @@
+                # Create a molecule object to hold the MM-optimized structures
+                compute.emm.append(energy)
+                compute.rmsd.append(rmsd)
+                if M_opts is None:
+                    M_opts = deepcopy(M_opt)
+                else:
+                    M_opts += M_opt
+            compute.emm = np.array(compute.emm)
+            compute.emm -= compute.emm[self.smin]
+            compute.rmsd = np.array(compute.rmsd)
+            if indicate:
+                if self.writelevel > 0:
+                    M_opts.write('mm_minimized.xyz')
+                    if self.ndim == 1:
+                        import matplotlib.pyplot as plt
+                        plt.switch_backend('agg')
+                        fig, ax = plt.subplots()
+                        dihedrals = np.array([i[0] for i in self.metadata['torsion_grid_ids']])
+                        dsort = np.argsort(dihedrals)
+                        ax.plot(dihedrals[dsort], self.eqm[dsort], label='QM')
+                        if hasattr(self, 'emm_orig'):
+                            ax.plot(dihedrals[dsort], compute.emm[dsort], label='MM Current')
+                            ax.plot(dihedrals[dsort], self.emm_orig[dsort], label='MM Initial')
+                        else:
+                            ax.plot(dihedrals[dsort], compute.emm[dsort], label='MM Initial')
+                            self.emm_orig = compute.emm.copy()
+                        ax.legend()
+                        ax.set_xlabel('Dihedral (degree)')
+                        ax.set_ylabel('Energy (kcal/mol)')
+                        fig.suptitle('Torsion profile: iteration %i\nSystem: %s' % (Counter(), self.name))
+                        fig.savefig('plot_torsion.pdf')
+            return (np.sqrt(self.wts)/self.energy_denom) * (compute.emm - self.eqm)
+        compute.emm = None
+        compute.rmsd = None
+
+        V = compute(mvals, indicate=True)
+
+        Answer['X'] = np.dot(V,V)
+
+        # Energy RMSE
+        e_rmse = np.sqrt(np.dot(self.wts, (compute.emm - self.eqm)**2))
+
+        self.PrintDict[self.name] = '%10s %10s    %6.3f - %-6.3f   % 6.3f - %-6.3f    %6.3f    %7.4f   % 7.4f' % (','.join(['%i' % i for i in self.metadata['torsion_grid_ids'][self.smin]]),
+                                                                                                                  ','.join(['%i' % i for i in self.metadata['torsion_grid_ids'][np.argmin(compute.emm)]]),
+                                                                                                                  min(self.eqm), max(self.eqm), min(compute.emm), max(compute.emm), max(compute.rmsd), e_rmse, Answer['X'])
+
+        # compute gradients and hessian
+        dV = np.zeros((self.FF.np,len(V)))
+        if AGrad or AHess:
+            for p in self.pgrad:
+                dV[p,:], _ = f12d3p(fdwrap(compute, mvals, p, p_idx = p), h = self.h, f0 = V)
```

Comment:
The argument `p_idx = p` can be removed here.

Reply:
Done.